server/plpgsql/statements.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plpgsql
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Statement represents a PL/pgSQL statement.
+type Statement interface {
+	// OperationSize reports the number of operations that the statement will convert to.
+	OperationSize() int32
+	// AppendOperations adds the statement to the operation slice.
+	AppendOperations(ops *[]InterpreterOperation, stack *InterpreterStack)
+}
+
+// Assignment represents an assignment statement.
+type Assignment struct {
+	VariableName  string
+	Expression    string
+	VariableIndex int32 // TODO: figure out what this is used for, probably to get around shadowed variables?
+}
+
+var _ Statement = Assignment{}
+
+// OperationSize implements the interface Statement.
+func (Assignment) OperationSize() int32 {
+	return 1
+}
+
+// AppendOperations implements the interface Statement.
+func (stmt Assignment) AppendOperations(ops *[]InterpreterOperation, stack *InterpreterStack) {
+	// TODO: figure out how I'm supposed to actually do this, rather than a search and replace
+	expression := stmt.Expression
+	var referencedVariables []string
+	for varName := range stack.ListVariables() {
+		if strings.Contains(expression, varName) {
+			referencedVariables = append(referencedVariables, varName)
+		}
```

Comment:
Ah... I see this is the issue you mentioned to me over chat... where `varname := 'input' || input;` incorrectly converts to `varname := '$1' || $2; `. It seems like we could follow up with a change this parsing to only replace bare identifiers – it just  means we have to be aware of quoting. If there's a tokenizer we could reuse to get the tokens in the expression, then maybe that's an easy way to slice it up?

Reply:
I see that `libpg_query` does expose the postgres scanner/lexer: https://github.com/pganalyze/libpg_query?tab=readme-ov-file#usage-scanning-a-query-into-its-tokens-using-the-postgresql-scannerlexer

`pg_query_go` has a `Scan()` function that we could potentially use for this.